Enterprise-CMCS/macpro-mdct-mcr
# [MCR] Extend NumberField masking to allow per-field rounding

### Description

Currently, we round every number field to 2 decimal places. Changing this so the default now becomes no rounding, and we can instead pass in specific rounding based on the field.

### Related ticket(s)


MDCT-2645

---
### How to test

- Login to MCR 
- Open MCPAR and open/add a submission
- Fill out number fields that both autosave and save on submit, confirm that the float value is being saved to the db and the comma-separated string is being displayed to the user
- Confirm that currency fields are rounded to 2 decimal places and other numeric fields are not rounding 
- Confirm that masking is working as expected
- Confirm that when reviewing PDF, values are comma-separated as needed
- Currency field experience:
    - https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/6752256/65bece0b-ea42-46c9-9f3b-f020fe54e7e1
- Default numeric field experience:
    - https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/6752256/d4999f05-79e4-4686-9151-43c266cea1b7







### Important updates




---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
